hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/response/TestCleanupTableInfo.java

```diff
@@ -66,4 +105,188 @@ public void checkAnnotationAndTableName() throws Exception {
       }
     });
   }
+
+  @Test
+  public void testHDDS4478() throws Exception {
+    HddsProtos.BlockID blockID = new BlockID(1, 1).getProtobuf();
+    String volume = "testVol";
+    String bucket = "testBuck";
+    String key = "/foo/bar/baz/key";
+
+
+    OMFileCreateRequest request =
+        anOmFileCreateRequest(blockID, volume, bucket, key);
+
+    OMMetadataManager omMetaMgr = createOMMetadataManagerSpy();
+    OMMetrics omMetrics = mock(OMMetrics.class);
+    OzoneManager om =
+        createOzoneManagerMock(volume, bucket, request, omMetaMgr, omMetrics);
+
+    OmVolumeArgs volumeArgs = aVolumeArgs(volume);
+    OmBucketInfo bucketInfo = aBucketInfo(volume, bucket);
+    addVolumeToMetaTable(volume, volumeArgs, omMetaMgr);
+    addBucketToMetaTable(volume, bucket, bucketInfo, omMetaMgr);
+
+    OzoneManagerDoubleBufferHelper dbh =
+        mock(OzoneManagerDoubleBufferHelper.class);
+
+    Map<String, Integer> cacheItemCount = new HashMap<>();
+    for (String tableName : omMetaMgr.listTableNames()){
+      cacheItemCount.put(tableName,
+          Iterators.size(omMetaMgr.getTable(tableName).cacheIterator()));
+    }
+
+
+    request.validateAndUpdateCache(om, 1, dbh);
+
+
+    CleanupTableInfo ann =
+        OMFileCreateResponse.class.getAnnotation(CleanupTableInfo.class);
+    List<String> cleanup = Arrays.asList(ann.cleanupTables());
+    for (String tableName : omMetaMgr.listTableNames()) {
+      if (!cleanup.contains(tableName)) {
```

Comment:
Here we have checked only tables which are not part of FileCreateResponse cleanupTable annotation.
Do we want to check tables which are affected also.

Just a question, not got what these lines are testing? (Is it just to see any tables which are not affected have same size in Cache) But how this is verifying fix, not sure if i am missing something basic here.

Reply:
Yes, the test does what you have summarized.
The basic idea is the following:
The issue is that we have certain epochs that are pushing entries to unexpected table caches. Unexpected in a way that eviction for those epochs on particular tables is not called when the DoubleBuffer flushes, because of the missing table name in the annotation.
I think it is sufficient to check whether we have added any unexpected cache entries to any other table's cache during applyTransaction. I might be wrong on this one, or there might be an easier way though.

Reply:
Ya, it makes sense to me. It is basically checking all other table cache entries should be what it has an initial value, which should not be changed.